packages/flutter/lib/src/widgets/binding.dart

```diff
@@ -106,16 +106,29 @@ abstract class WidgetsBindingObserver {
   /// [SystemChannels.navigation].
   Future<bool> didPushRoute(String route) => Future<bool>.value(false);
 
-  /// Called when the host tells the app to push a new route information onto
-  /// the router.
+  /// Called when the host tells the application to push a new
+  /// [RouteInformation] and a restoration state onto the router.
+  ///
+  /// The [restorationData] contains the restoration information that the
+  /// platform wants this application to restore if this application opt for
+  /// state restoration. Otherwise, The [restorationData] will be null.
   ///
   /// Observers are expected to return true if they were able to
   /// handle the notification. Observers are notified in registration
   /// order until one returns true.
   ///
   /// This method exposes the `pushRouteInformation` notification from
-  /// [SystemChannels.router].
-  Future<bool> didPushRouteInformation(String route, Object state) => Future<bool>.value(false);
+  /// [SystemChannels.navigation].
+  ///
+  /// The default implementation is to call the [didPushRoute] directly with the
+  /// input string [location].
+  Future<bool> didPushRouteInformation(
```

Comment:
Maybe `didPushRouteInformation` should be given a RouteInformation object as an argument. That way, we can easily add more data to this message if necessary. Basically, I am thinking:

1. move RouteInformation class to binding.dart
2. replace the arguments of this method with a routeinformaiton object
3. remove the restorationdata for now and once we know what type that would have we can easily add it back into the object.